cpp/include/cuspatial/detail/utility/device_atomics.cuh

```diff
@@ -14,6 +14,61 @@
  * limitations under the License.
  */
 
+#pragma once
+
+#include <algorithm>
+#include <type_traits>
+
+namespace {
+
+template <typename T>
+const T& __device__ min(const T& a, const T& b)
+{
+  return std::min(a, b);
+}
+
+template <typename T>
+const T& __device__ max(const T& a, const T& b)
+{
+  return std::max(a, b);
+}
+
+template <typename T,
```

Comment:
This needs docs, especially for all the template parameters.

Reply:
See updated docs.